config/puma/production.rb

```diff
@@ -1,3 +1,30 @@
+require 'English'
+
+# Puma will only listen on what it's configured to listen on. When running
+# under systemd socket activation, the configuration where to listen is in
+# foreman.socket. This code translates whatever listen file descriptors it
+# received from systemd into bind statements. It also clears anything it was
+# configured on.
+if ENV['LISTEN_FDS'] && ENV['LISTEN_PID'].to_i == $PID
```

Comment:
Looks like `LISTEN_FDS` is not in `ENV` at this point. Digging further.

Reply:
Looks like the `Binder` clears the variable from the env. See https://github.com/puma/puma/issues/1360#issuecomment-689548650 for more details.